core/src/test/java/com/linecorp/armeria/client/HttpResponseDecoderTest.java

```diff
@@ -107,4 +111,17 @@ void confirmResponseStartAndEndInTheSameThread(SessionProtocol protocol)
         latch.await(System.getenv("CI") != null ? 60 : 10, TimeUnit.SECONDS);
         assertThat(failed.get()).isFalse();
     }
+
+    @ParameterizedTest
+    @EnumSource(value = SessionProtocol.class, names = {"H1C", "H2C"})
+    void responseCompleteNormallyIfRequestIsAborted(SessionProtocol protocol) throws Exception {
```

Comment:
maybe some more assertions are needed here?

i don't see where the request is aborted (could be a comment if I'm just not seeing it)

also, might need a signal to ensure the `exceptionally` was actually invoked, as I would guess we are assuming that hook is in fact called. If it weren't the assertion would silently not run

Reply:
That's a good point.
Changed to use `handle` which is always executed and add the comment. 😄

Reply:
indeed it is addressed!